### PR TITLE
working but must be in the path before harness/rest-server/bin or its…

### PR DESCRIPTION
**Pull request by pferrel**
… symlink

### CODE REVIEW

File: python-cli/common.sh [hunk 1]
```diff
@@ -13,26 +13,45 @@ export CYAN='\033[0;36m'         # Cyan--hints and info messages
...
+if [ -z "${HARNESS_CLI_HOME}" ]; then
+  if [ -z "$(which readlink 2>/dev/null)" ]; then
+    echo -e "${RED}readlink command must be present on your system!${NC}"
+    exit 1
   fi
+
+  #bindir=$(dirname `_readlink_f $0`)
+  # trim the last path element (which is supposed to be /bin(/))
+  #export $HARNESS_CLI_HOME=${bindir%/*}
+  export HARNESS_CLI_HOME=$(dirname `_readlink_f $0`)
```
Comment (pferrel) [inline python-cli/common.sh]: @dennybaa 

I have a symlink:
```sudo ln -s /Users/pat/harness-cli/python-cli /usr/local/harness-cli```
The symlink is first in my PATH before the embedded harness cli

This works on macOS, not sure why you are trimming the last subdir off the env var. In this case HARNESS_CLI_HOME would be `/Users/pat/harness-cli` so every reference to a file to be sourced or executed would have to be prefixed with `/python-cli` to make `${HARNESS_HOME}/python-cli/common.sh` and this clearly makes no sense so I removed the trimming for this CLI but now I wonder if this does not work on Linux.

I also do not understand why the code DOES work for the Makelfile build of harness on macOS and linux. I bet it is broken now for the script build of harness with `make-harness-distribution.sh`. 

The env var `HARNESS_HOME` or `HARNESS_CLI_HOME` should be the full physical filesystem path (no symlink) to the `bin` directory of harness or the `python-cli` directory of this harness-cli -- right? BTW I still use your `_readlink_f` but we must have different ideas for where project HOME points.

What do you think the value of `HARNESS_HOME` should be?

Comment (pferrel) [reply python-cli/common.sh]: Maybe our problem is that common.sh is sourced twice and calculates completely different values for HARNESS_CLI_HOME here and also for the embedded cli

**Comment by pferrel** [comment]
common.sh was being sourced twice, refactored into common-env.sh which is sourced only once and common-functions.sh which is sourced in all command scripts to ge access to function definitions.

hacky but will be gone when re-written in python.

File: python-cli/common.sh (hunk 1 above)
Comment (dennybaa) [reply python-cli/common.sh]: The initial script aimed to locate the `HARNESS_HOME` which is not required in this case. Exporting `HARNESS_CLI_HOME` seems should not happen at all. Splitting functions and common into two files is an overkill too.

**Comment by pferrel** [comment]
I left some called or sourced files named the same as embedded CLI so the use of `HARNESS_CLI_HOME` guarantees the right scripts are used. Trimming the last dir from the readlink was removed here and it has been a problem in the embedded CLI so it still puzzles me but I don't care enough to deal with it anymore.

This works so moving to develop, then I'll test on ubuntu

**Pull request merged**